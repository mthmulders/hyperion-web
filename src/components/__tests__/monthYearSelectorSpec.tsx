import * as React from 'react';

import renderer from 'react-test-renderer';

import MonthYearSelector from '../monthYearSelector';

describe('<MonthYearSelector />', () => {
    it('should look acceptable', () => {
        // Act
        const wrapper = renderer.create(
            <MonthYearSelector
                selectedMonth={ 2018 }
                selectedYear={ 1 }
                updateSelectedMonth={ jest.fn() }
                updateSelectedYear={ jest.fn() }
            />,
        );

        // Assert
        expect(wrapper.toJSON()).toMatchSnapshot();
    });
});
