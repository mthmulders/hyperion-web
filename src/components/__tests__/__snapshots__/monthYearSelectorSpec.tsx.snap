// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MonthYearSelector /> should look acceptable 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MonthYearSelector
    selectedMonth={2018}
    selectedYear={1}
    updateSelectedMonth={[MockFunction]}
    updateSelectedYear={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <MonthSelector
          selectedMonth={2018}
          updateSelectedMonth={[MockFunction]}
        />,
        <br />,
        <YearSelector
          selectedYear={1}
          updateSelectedYear={[MockFunction]}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "selectedMonth": 2018,
          "updateSelectedMonth": [MockFunction],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "selectedYear": 1,
          "updateSelectedYear": [MockFunction],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <MonthSelector
            selectedMonth={2018}
            updateSelectedMonth={[MockFunction]}
          />,
          <br />,
          <YearSelector
            selectedYear={1}
            updateSelectedYear={[MockFunction]}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "selectedMonth": 2018,
            "updateSelectedMonth": [MockFunction],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "selectedYear": 1,
            "updateSelectedYear": [MockFunction],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
